articles/Frontend/Architecture/coding-guide.md

```diff
@@ -380,32 +457,57 @@ countdown
 ```
 
 TAO provides an implementation, the `core/eventifier` module, it has the following features :
- - contextualized to an object (events are scoped)
- - support [AOP](https://en.wikipedia.org/wiki/Aspect-oriented_programming) style listening (`before` -> `on` -> `after`)
- - support namespaces
- - support `Promise` (asynchronous handlers)
- - supports context spreading
+
+- contextualized to an object (events are scoped)
+- support [AOP](https://en.wikipedia.org/wiki/Aspect-oriented_programming) style listening (`before` -> `on` -> `after`)
+- support namespaces
+- support `Promise` (asynchronous handlers)
+- supports context spreading
 
 Please check out the [eventifier documentation](#eventifier).
 
 ### Provider
 
 When multiples implementation of a given API can be defined, or dynamically defined, the provider pattern is used.
 
-_TBD_
+```js
+const jsonDataProvider = {
+  requestData() {
+    //read some json file and return data
+  },
+};
+
+const csvDataProvider = {
+  requestData() {
+    //read some csv file and return data
+  },
+};
+
+const person = {
+  dataProvider,
+  data,
+  updateData() {
+    this.data = this.dataProvider.requestData();
+  },
+  registerProvider(provider) {
+    this.dataProvider = provider;
+  },
+};
+```
+
+In this example two providers are defined, which have same `requestData` API method. `person` object does not request data directly and doesn't implement any data parsing logic, it just registers the appropriated provider and request data from it. This allows to decouple components, keep components small and easy testable.
```

Comment:
**chore:** You could also mention it works well together with the delegation pattern.

Reply:
Added few words on the topic, thanks for idea